internal/function_time_parser.go

```diff
@@ -518,6 +518,40 @@ func centuryFormatter(t *time.Time) ([]rune, error) {
 	return []rune(fmt.Sprint(t.Year())[:2]), nil
 }
 
+func yearWithoutCenturyParser(text []rune, t *time.Time) (int, error) {
+	const yearLen = 2
+	if len(text) > yearLen {
```

Comment:
This clause is incorrect for the case of `01-MM-DD`. I will need to update this.

A related issue is Recidiviz/recidiviz-data#20756

BigQuery generally allows non-zero leading dates, but the emulator does not.

Reply:
This has been updated to fix goccy/go-zetasqlite#137